### ISSUE TITLE
Win10 - Python3.6.6 + PyQT5 (5.15.6) + cx_Freeze 6.10 : Qt5Core.dll dependency with python3.dll(?!) not found

### ISSUE THREAD

**Issue by LoneWanderer-GH** [marcelotduarte/cx_Freeze#1408]
I recently faced an issue with cx_Freeze and PyQt5, that seems somewhat related to internal guts of the generated exe (sys.path management ? the way dll copies are made ?)

This happens with my project, and also with the sample project included in cx_Freeze repository.

It is somewhat related to this stackoverflow question https://stackoverflow.com/questions/58018779/create-executable-with-python-3-7-pyqt5-and-cx-freeze-but-dll-failed-to-load (which concerns an older version of cx_Freeze and references https://github.com/anthony-tuininga/cx_Freeze/issues/504)

### Workaround (long story short)

At first I thought it was related to some Microsoft libs (vcruntime140.dll or msvc++ redist and stuff like that).

In the end the following worked (found by trial and error) :

Copy either `python3.dll` or `python36.dll` into `lib/PyQt5`
python3.dll is preferrable since its the smallest one (~58 Kibi)

Tools such a Dependency Walker did not helped whatsoever ...

### Steps to reproduce

1. on a win10 machine, with python 3.6.6 installed and added to the PATH
2. Create a project with the cx_Freeze sample code for PyQt5, when performing necessary pip commands, the venv spec should look like:
```pip
cx-Freeze==6.10
cx-Logging==3.0
importlib-metadata==4.8.3
lief==0.11.5
PyQt5==5.15.6
PyQt5-Qt5==5.15.2
PyQt5-sip==12.9.1
typing-extensions==4.1.1
zipp==3.6.0
```
note: you can remove/comment the following lines to have the errors print to console for easier reading/analysis
```python
if sys.platform == "win32":
    base = "Win32GUI"

```
3. Execute the setup script. You should obtain an executable in
`<project>/build/exe.win-amd64-3.6/test_pyqt5.exe`

4. Launch the excutable : it may start correctly
5. Create a DOS batch script as follows :
```bat
@ECHO OFF
REM emulate usage on a machine without python3+ installed
SET PATH=modify it to REMOVE all references to python3+ installs
PUSHD build\exe.win-amd64-3.6
test_pyqt5.exe
POPD
```
6. The app won't start and a popup will state that a PyQt5 DLL could not be found. The console version of this would display something like this:

```
Traceback (most recent call last):
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\Lib\site-packages\cx_Freeze\initscripts\__startup__.py", line 113, in run
    module_init.run(name + "__main__")
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\Lib\site-packages\cx_Freeze\initscripts\Console.py", line 15, in run
    exec(code, module_main.__dict__)
  File "PyQt5app.py", line 5, in <module>
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\lib\site-packages\PyQt5\__init__.py", line 54, in <module>
ImportError: DLL load failed: Le module spécifié est introuvable. (<= localized error message, but you get it)
```

The code corresponds to `PyQt5.__ini__.py` trying to import `Qt5Core.dll`
The workaround indicated at the beginning solves the problem.

### Final remark

I also noticed that the produced exe contains `/lib/PyQt5/Qt5/bin/` folder with a whole bunch of DLLs (~80 Mibi).
If I delete the folder, the app works just fine ...

**Comment by LoneWanderer-GH** [comment]
See in particular this reply to #504

https://github.com/marcelotduarte/cx_Freeze/issues/504#issuecomment-530398554

**Comment by marcelotduarte** [comment]
For me the samples works!
I do not read all of this thread yet, but I think that I know the issue.
Please check the [FAQ](https://cx-freeze.readthedocs.io/en/latest/faq.html#microsoft-visual-c-redistributable-package).

**Comment by marcelotduarte** [comment]
Another possibility is the use o lief in 6.10. It is more aggressive to detect the dependencies and I must check this (Similar issue #1372).
In PR https://github.com/marcelotduarte/cx_Freeze/pull/1344 there are instructions to disable this experimental behavior.

**Comment by LoneWanderer-GH** [comment]
The important thing on my side was removing all python stuff from path/environment variables before launching the exe.

---

 - Using `"include_msvcr": True,` does not change the observed behaviour (it indeed adds several dlls, but the problem persists)

 - setting an env. variable `CX_FREEZE_BIND=imagehlp` does not change the observed behaviour

 - The following include directive does not work, due to `python3.dll` being on an internal exception list. So the copy should be hardcoded at the end of the process.

```python
python_dir = Path(sys.executable).parent
python_dlls = [ Path(p) for p in glob(f"{python_dir.as_posix()}/python3.dll")]
pyqt_base_dir = Path("lib", "PyQt5")
# prepare cx_Freeze tuples (source file, target dir)
includes_list+= [ (p, pyqt_base_dir / p.name) for p in python_dlls ]
```

---

Here is the build log with all files, and generated directory tree:

[build_log_without_lief.txt](https://github.com/marcelotduarte/cx_Freeze/files/8223760/build_log_without_lief.txt)
[build_log_with_lief.txt](https://github.com/marcelotduarte/cx_Freeze/files/8223761/build_log_with_lief.txt)

[tree_without_lief.txt](https://github.com/marcelotduarte/cx_Freeze/files/8223870/tree_without_lief.txt)
[tree_with_lief.txt](https://github.com/marcelotduarte/cx_Freeze/files/8223871/tree_with_lief.txt)

Generated tree diff between lief and legacy (which does not really help):

<img width="574" alt="image" src="https://user-images.githubusercontent.com/19716859/157675462-416541bd-8ffe-408b-96f2-5a443b435cad.png">

**Comment by marcelotduarte** [comment]
In PR I didn't explain, so just to confirm:
You must set CX_FREEZE_BIND=imagehlp before the build.

**Comment by LoneWanderer-GH** [comment]
> In PR I didn't explain, so just to confirm: You must set CX_FREEZE_BIND=imagehlp before the build.

I confirm, this is set as en environment variable right before build.

**Comment by marcelotduarte** [comment]
I used python 3.6.8 from GHA to test. Can you upgrade and test?

**Comment by LoneWanderer-GH** [comment]
I'm sorry but I can't test 3.6.8 due to corporate limitations (no admin rights).
---
Here are some clarified tests using the example.

All runs are made from within a clean new console, with all possible python.exe/python dlls removed from `%PATH%` environment variable.

Same requirements.txt is used for all python versions:

```
cx-Freeze==6.10
cx-Logging==3.0
importlib-metadata==4.8.3
lief==0.11.5
PyQt5==5.15.6
PyQt5-Qt5==5.15.2
PyQt5-sip==12.9.1
typing-extensions==4.1.1
zipp==3.6.0
```

Modifications to the example `setup.py` are as follows:

```python
# base="Win32GUI" should be used only for Windows GUI app
base = None
# use console mode instead of pure GUI for easier stack trace copy !
# if sys.platform == "win32":
#     base = "Win32GUI"
```

and 

```python
import sysconfig
# ...
build_dir = Path("build") # new
# use CX_FREEZE_BIND env var if defined, otherwise put 'with_lief' in exe build dir name
platform_ident = f"exe.{sysconfig.get_platform()}-{sysconfig.get_python_version()}-{os.getenv('CX_FREEZE_BIND', 'with_lief')}" # new

build_exe_options = {
    "build_exe": build_dir / platform_ident, # new
    "excludes": ["tkinter"],
    "include_files": include_files,
    "include_msvcr": True,
    "silent_level": 0,
}
```

Build                                            | Python version  | CX_FREEZE_BIND | Status | build log
-----------------------------------|-------------------|--------------------|----------|---
exe.win-amd64-3.6-imagehlp   | 3.6.6                   |  imagehlp | ❌ | [build_3.6_no_lief.log](https://github.com/marcelotduarte/cx_Freeze/files/8375155/build_3.6_no_lief.log)
exe.win-amd64-3.6-with_lief     | 3.6.6                   |  _not set_     | ❌ | [build_3.6_lief.log](https://github.com/marcelotduarte/cx_Freeze/files/8375159/build_3.6_lief.log)
exe.win-amd64-3.9-imagehlp   | 3.9.6                   |  imagehlp | ✔️ | [build_3.9_no_lief.log](https://github.com/marcelotduarte/cx_Freeze/files/8375158/build_3.9_no_lief.log)
exe.win-amd64-3.9-with_lief     | 3.9.6                   |  _not set_     | ✔️ | [build_3.9_lief.log](https://github.com/marcelotduarte/cx_Freeze/files/8375156/build_3.9_lief.log)

Trace for python 3.6.6 is always the same in both cases:
```
Traceback (most recent call last):
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\Lib\site-packages\cx_Freeze\initscripts\__startup__.py", line 113, in run
    module_init.run(name + "__main__")
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\Lib\site-packages\cx_Freeze\initscripts\Console.py", line 15, in run
    exec(code, module_main.__dict__)
  File "PyQt5app.py", line 5, in <module>
  File "C:\Users\REDACTED\cx_Freeze_PyQt5_Test\venv\lib\site-packages\PyQt5\__init__.py", line 54, in <module>
ImportError: DLL load failed: Le module spécifié est introuvable.
```

So this might me something specific to python 3.6.6 ...
Maybe keep this ticket open for future references, or if anyone encounters this issue with other intermediate python versions.

And maybe close it once #1382  happens

**Comment by marcelotduarte** [comment]
As it worked for me in a different version than yours, I thought it might be a python problem. Then I found a bug that apparently applies.
https://bugs.python.org/issue29778

I asked you to test the version above for confirmation only. But I intended to recommend you to use py38 or py39. If you are going to use py39 this is very good. Because in python 38+ windows there is a change in the way of fetching the dlls and pyqt5 and pyside2 do this search correctly. In previous versions it was very messy. ;-)
